### DISCUSSION TITLE
Matomo Tracking-Code

### DISCUSSION THREAD

**Discussion by mjegulla** [orestbida/cookieconsent#279] [Q&A] [answered]
Does it work with Matomo Tracking code?
Thank you

**Comment by orestbida** [comment, marked as answer]
Hi @mjegulla,

yes, but you need to configure the tracking code appropriately. Check out [matomo's docs](https://developer.matomo.org/guides/tracking-consent).

Here is one way you could implement it:
1. Place matomo's the tracking code into your pages (added `_paq.push(['requireConsent'])`):
    ```html
    <!-- Matomo -->
    <script type="text/javascript">
        var _paq = window._paq = window._paq || [];
        // require user tracking consent before processing data
        _paq.push(['requireConsent']);
        _paq.push(['trackPageView']);
        _paq.push(['enableLinkTracking']);
        (function() {
            var u="//{$MATOMO_URL}/";
            _paq.push(['setTrackerUrl', u+'matomo.php']);
            _paq.push(['setSiteId', {$IDSITE}]);
            var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
            g.type='text/javascript'; g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
        })();
    </script>

2. Configure the plugin (cookieconsent-init.js)
    ```javascript
    
    const cc = initCookieConsent();
    
    cc.run({
        // your config options ...
    
        onAccept: function(){
            if(cc.allowedCategory('analytics')){
                _paq.push(['setConsentGiven']);
            }
        },
        
        onChange: function(){
            if(cc.allowedCategory('analytics') === false){
                _paq.push(['forgetConsentGiven']);
            }
        },
    
        // more config options ...
    });
    ```

You'll need to define the 'analytics' category inside the plugin for this to work. Also note that the above code will fully disable matomo's tracking (and not just cookies) if the user doesn't accept the analytics category.

**Reply by mjegulla** [reply]
Wow, thanks a lot for your fast reply and help.
I managed to come half way through. 
To define the 'analytics' category inside the plugin is probably out of my capacity 🤷🏻‍♂️.

**Reply by orestbida** [reply]
A category is defined whenever you see a `toggle` object.

E.g. the toggle object in the demo example below, is what defines the `analytics` category.
https://github.com/orestbida/cookieconsent/blob/8a5fd66f94f884413331454f5b735e9b7953c00f/demo/demo_gtm/cookieconsent-init.js#L97-L103

**Reply by mjegulla** [reply]
Thank you 🙏

**Reply by Brumonti** [reply]
Exactly my problem for the past 2 days and now it is solved. Thanx!!!
